vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -408,17 +464,28 @@ impl<T: DeviceBacking> NvmeDriver<T> {
     }
 
     /// Gets the namespace with namespace ID `nsid`.
-    pub async fn namespace(&self, nsid: u32) -> Result<Namespace, NamespaceError> {
-        Namespace::new(
-            &self.driver,
-            self.admin.as_ref().unwrap().clone(),
-            self.rescan_event.clone(),
-            self.identify.clone().unwrap(),
-            &self.io_issuers,
-            &self.device_id,
-            nsid,
-        )
-        .await
+    pub async fn namespace(&mut self, nsid: u32) -> Result<Arc<Namespace>, NamespaceError> {
+        // Check if namespace was already added after restore.
+        if let Some(ns) = self.namespace.iter().find(|n| n.nsid() == nsid) {
```

Comment:
This will break the case where the namespace is hot removed and hot added back with a different identity.

Reply:
Not sure I follow. In vtl2 settings we identify controller's namespace by pci id + namespace id.
Do you expect to have dynamic vtl2 settings in future?

Reply:
We discussed this offline, so sharing here. The existence of namespaces is controlled by hardware. And, namespaces can come and go at runtime. It's possible that a namespace has disappeared while the OpenHCL servicing operation. (Or, worse: disappeared and a new one came back!)

Reply:
Re-querying the namespace for now. Namespace save/restore code remains but is inactive.